### PR TITLE
Wire in steps for entering a locked room

### PR DESCRIPTION
**Pull request by user512**
* [x] Fix AccessLevels getting undefined values from the accessLevel parameterType
* [X] Implement step definition for entering a workspace with a Locked Room
* [X] Implement step definition for entering a Locked Room
* [X] Implement step definition for being prevented from entering a Locked Room
* [X] Expose the Room's Access Level in the the room card's DOM
* [X] Add helper method to Workspace PageObject for selecting RoomCards from the DOM
* [x] Verify that an error message is provided when entering the wrong room and that people don't enter the room
* [ ] (Maybe?) Clarify parameter type and regex in README (@KellyAH, @zspencer)
* [ ] (Maybe?) Align SystemTest Workspace locked room access code with Demo Workspace

### CODE REVIEW

**Comment by zspencer** [comment]
Would you be able to amend the commit to include CJ, Egbet, Kelly and my Github Emails?

```
Co-authored-by: CJ Joulain <cjoulain@users.noreply.github.com>
Co-authored-by: Zee Spencer <zspencer@users.noreply.github.com>
Co-authored-by: KellyAH <KellyAH@users.noreply.github.com>
Co-authored-by: Egbet <eg-bet@users.noreply.github.com>
```

File: convene-web/app/views/workspaces/_room_card.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<li class="col-span-1 bg-white rounded-lg shadow mb-4" id="<%= room.name %>" >
+<li class="col-span-1 bg-white rounded-lg shadow mb-4 <%= "--#{room.access_level}" %>" id="<%= room.name %>">
```
Comment (zspencer) [inline convene-web/app/views/workspaces/_room_card.html.erb:1]: Not-blocking: Something that may be nice would be to also include a `room-card` class to the `li` so we can key off of that as well.

File: convene-web/app/models/room.rb [hunk 2]
```diff
@@ -19,6 +19,7 @@ class Room < ApplicationRecord
   # `locked` indicates that only participants who know the rrooms `access_code` may access the room.
   # `internal` indicates that only participants who are Workspace Members _or_ know the Workspaces `access_code` may
   # access the room.
+  # TODO Return :unlocked by default, add database non null
```
Comment (zspencer) [inline convene-web/app/models/room.rb:22]: Let's do this in it's own PR

File: features/page-objects/WorkspacePage.js [hunk 3]
```diff
@@ -48,6 +48,11 @@ class WorkspacePage extends Page {
     assert.equal(videoPanels.length, 1, `${videoPanels.length} was found.`)
     return await this.driver.findElement(jitsiConferenceFrame);
   }
+
+  roomCardsWhere({ accessLevel }) {
+    const selector = `.--${accessLevel.level.toLowerCase()}`
+    return this.driver.findElements(By.css(`.--${accessLevel.level.toLowerCase()}`));
```
Comment (zspencer) [inline features/page-objects/WorkspacePage.js:54]: Not blocking, but one way we could start to distribute responsibility more broadly would be to move the responsibility for knowing how to find the accessLevel element into the other parameterTypes. I think `roomCardsWhere` still makes sense on the Workspace object, and doing the actual `driver.findElements()` within it feels good to me.

```
// in parameterTypes/workspacePage.js
class WorkspacePage {
  // ...
  roomCardsWhere({ accessLevel }) {
    return this.driver.findElements(accessLevel.selector))
  }
  // ...
}

// in parameterTypes/accessLevel.js
class AccessLevel {
  // See the MDN Getter docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get
  get selector() {
    return By.css(`.--${this.level.toLowerCase()}`)
  }
}
```

File: features/parameter-types/rooms.js [hunk 4]
```diff
@@ -33,8 +33,8 @@ class Room {
 // See: https://github.com/zinc-collective/convene/issues/41
 defineParameterType({
   name: "accessLevel",
-  regexp: /(an |a )?(Unlocked|Internal|Locked)/,
-  transformer: (level) => new AccessLevel(level),
+  regexp: /(an |a |the )?(Unlocked|Internal|Locked)/,
```
Comment (zspencer) [inline features/parameter-types/rooms.js:36]: Not blocking: We may want to remove the `?` from this regex, so that we don't have to deal with different numbers of arguments being passed into the Transformer.

File: features/steps/room_steps.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
...
+const Workspace = require("../parameter-types/workspaces");
+const Room = require("../parameter-types/rooms");
 const assert = require('assert').strict;
 
-Given("a Workspace with {accessLevel} {room}", function (accessLevel, room) {
-  // Write code here that turns the phrase above into concrete actions
-  return "pending";
+Given("a Workspace with {accessLevel} {room}", async function (accessLevel, room) {
+  const workspace = new Workspace("System Test");
+  this.workspace = new WorkspacePage(this.driver, workspace);
```
Comment (zspencer) [inline features/steps/room_steps.js:9]: Not blocking, something to think about is we have a WorkspacePage and a Workspace object and I'm not sure where the boundaries between these two things are. Does it make sense for there to be two different classes? Or could we do something to add the `driver` to the `Workspace` class? I'm not really sure. Something to noodle on over the next few months.

File: features/steps/room_steps.js [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
+  await this.workspace.enter();
+  const lockedRooms = await this.workspace.roomCardsWhere({ accessLevel });
```
Comment (zspencer) [inline features/steps/room_steps.js:11]: Not blocking: I almost wish we had a `fluent interface` for this. I.e. `await this.workspace.enter().roomCardsWhere({ accessLevel })` but I think that may be hard to pull off.

**Review by zspencer — COMMENTED** [review]
I think this is a really incredible first step! I think there's some efforts that may (or may not) be worth making to place responsibility more squarely where it belongs; but in the meantime I think once we get the assertion in about the error message and (maybe) figure out a way to verify the video panel is not visible we're gonna be good-to-go.

**Comment by zspencer** [comment]
Also, could you add a reference to the GithubIssue we are working on:

```
See: https://github.com/zinc-collective/convene/issues/40
```

File: convene-web/app/views/workspaces/_room_card.html.erb (hunk 1 above)
Comment (user512) [reply convene-web/app/views/workspaces/_room_card.html.erb:1]: 👍 Will be working on this on a separate PR.

File: features/page-objects/WorkspacePage.js [hunk 7]
```diff
@@ -48,6 +50,11 @@ class WorkspacePage extends Page {
     assert.equal(videoPanels.length, 1, `${videoPanels.length} was found.`)
     return await this.driver.findElement(jitsiConferenceFrame);
   }
+
+  roomCardsWhere({ accessLevel }) {
+    const selector = `.--${accessLevel.level.toLowerCase()}`
```
Comment (user512) [inline features/page-objects/WorkspacePage.js:55]: This line is dead code.

File: features/page-objects/WorkspacePage.js (hunk 3 above)
Comment (user512) [reply features/page-objects/WorkspacePage.js:55]: I like this.
